projects/kserve/visualizations/kserve-llm/models/kpi.py

```diff
@@ -0,0 +1,89 @@
+import matrix_benchmarking.models as matbench_models
+
+#
+# lts.VERSION must be bumped each time a KPI is added
+#
+
+KPIs = {} # populated by the @matbench_models.KPIMetadata decorator
+
+# Throughput
+
+@matbench_models.KPIMetadata(help="Model throughput", unit="tokens/s")
+def kserve_llm_load_test_throughput(lts_payload):
+    return lts_payload.results.throughput
```

Comment:
Hey @drewrip what does `lts_payload` here refers to? Is it the llm-load-test output json or something else?

Reply:
This is the raw Python object that is being assembled before it is parsed by Pydantic. You can see this object being assembled in this file: https://github.com/openshift-psap/topsail/blob/main/projects/kserve/visualizations/kserve-llm/store/lts_parser.py

It collects metrics, data and info from a number of sources.

Reply:
yes, currently it's done on multiple steps, for historical reasons:
llm-load-test output -> [matrix-benchmarking entry](https://github.com/openshift-psap/topsail/blob/693d2f4cb9375fc2f86b87ee6a6f8af28cb97e0a/projects/kserve/visualizations/kserve-llm/store/parsers.py) --> [LTS entry](https://github.com/openshift-psap/topsail/blob/693d2f4cb9375fc2f86b87ee6a6f8af28cb97e0a/projects/kserve/visualizations/kserve-llm/store/lts_parser.py) --> [KPI entries](https://github.com/openshift-psap/topsail/blob/693d2f4cb9375fc2f86b87ee6a6f8af28cb97e0a/projects/kserve/visualizations/kserve-llm/models/kpi.py)